src/minerva_lib/blend.py

```diff
@@ -1,102 +1,87 @@
+            }
+
+    Returns:
+        Threshholded float32 image normalzied within 0, 1
+        r, g, b float32 array color within 0, 1
     '''
 
-    # No well-defined behavior for decimal values or values less than 0
-    try:
-        dtype_info = np.iinfo(img.dtype)
-        assert dtype_info.min is 0
-    except (ValueError, AssertionError):
-        raise ValueError('Scaling to 0,1 requires unsigned integers')
+    f32_range = (channel['min'], channel['max'])
+    f32_image = skimage.img_as_float(channel['image'])
 
-    one = dtype_info.max
-    return np.float32(img / one)
+    # Return rescaled image and color for additive blending
+    f32_image = skimage.exposure.rescale_intensity(f32_image, f32_range)
+    return f32_image, channel['color']
 
 
-def f32_to_bgr(f_img, color=[1, 1, 1]):
-    '''Reshape into a color image
+def make_rgb(image, color):
+    ''' Yield 3 image channels for r, g, b
 
     Arguments:
-    f_img: float32 image to reshape
+        image: Numpy 2D image data
+        color: r, g, b float32 array within 0, 1
+
+    Yields:
+        Numpy 2D image data for r, g, b channels
     '''
 
-    # All inputs should be normalized between 0 and 1
-    try:
-        assert np.all((f_img >= 0) & (f_img <= 1))
-    except AssertionError:
-        raise ValueError('Color image requires values from 0,1')
+    for scalar in color:
+        yield image * scalar
 
-    # Give the image a color dimension
-    f_vol = f_img[:, :, np.newaxis]
-    f_bgr = np.repeat(f_vol, 3, 2) * color
-    return np.round(255 * f_bgr).astype(np.uint8)
 
+def linear_rgb(channels):
+    '''Blend all channels into one normalized image
 
-def linear_bgr(channels):
-    '''Blend all channels given
     Arguments:
         channels: List of dicts of channels to blend with rendering settings:
             {
                 image: Numpy image data
-                color: N-channel by b, g, r float32 color
-                min: Range minimum, float32 range
-                max: Range maximum, float32 range
+                color: r, g, b float32 array within 0, 1
+                min: Range minimum, float32 range 0, 1
+                max: Range maximum, float32 range 0, 1
             }
 
     Returns:
-        uint8 y by x by 3 color BGR image
+        float32 y by x by r, g, b color image within 0, 1
     '''
 
-    # Get the number of channels
-    num_channels = len(channels)
-
-    # Must be at least one channel
-    if num_channels < 1:
-        raise ValueError('At least one channel must be specified')
-
-    # Ensure that dimensions of all channels are equal
-    shape = channels[0]['image'].shape
-    for channel in channels:
-        if channel['image'].shape != shape:
-            raise ValueError('All channel images must have equal dimensions')
-
-    # Shape of 3 color image
-    shape_color = shape + (3,)
-
-    # Final buffer for blending
-    image_buffer = np.zeros(shape_color, dtype=np.float32)
+    num_channels = 0
```

Comment:
Just move the check for 0 channels to here. Suggest:

```python
if not channels:
    ...
```

Reply:
Also at this point check that the dimensions of all the channels are the same.

Reply:
I set up the check this way to allow for `channels` to be any iterable (not only a list). While `not []` is true, `not iter([])` is false. It's still possible to check that all dimensions are the same with an iterator, but we couldn't do anything upfront like this on an iterator.

If we don't want to support both lists and iterators, that's okay. I have not actually confirmed that loading numpy arrays iteratively is more performant than loading them all into a list. Perhaps numpy arrays are only actually loaded into memory at the moment the data are needed.

Reply:
done `refactor composite_image for single channel`